delocate/libsana.py

```diff
@@ -55,9 +55,12 @@ def tree_libs(start_path, filt_func=None):
                 continue
             rpaths = get_rpaths(depending_libpath)
             for install_name in get_install_names(depending_libpath):
-                lib_path = (install_name if install_name.startswith('@')
-                            else realpath(install_name))
-                lib_path = resolve_rpath(lib_path, rpaths)
+                # If the library starts with '@rpath' we'll try and resolve an rpath
```

Comment:
Install name can also start with `@loader_path`

https://matthew-brett.github.io/docosx/mac_runtime_link.html

Presumably we don't want to run this code in that case?  `if install_name.startswith('@rpath`)` ?

Sorry if I forgot - but did you conclude that the system searches for libs via the environment variables, when the install name has @rpath prefix?

Reply:
`resolve_rpath()` does its own check to see if `install_name` starts with `'@rpath'` -- it would absolutely work to move that up here if you like. I had been trying to keep the code the way you had it before (where you only checked to see if there was an `@` -- see line 58).

It's been a little while since I investigated it, but I did find (https://github.com/matthew-brett/delocate/pull/26#issuecomment-369306639) that the system will use `DYLD` and `DYLD_FALLBACK` to resolve **@rpath** when searching for libraries.

Reply:
Hokay, make it a bit more explicit about what's happening. I've also double-checked and confirmed that the system uses the three environment variables as search options for `@rpath`.